models/gif_psc_exp.cpp

```diff
@@ -294,26 +290,21 @@ nest::gif_psc_exp::calibrate()
   // since t_ref_ >= 0, this can only fail in error
   assert( V_.RefractoryCounts_ >= 0 );
 
-  // initializing internal state
-  if ( !S_.sfa_stc_initialized_ )
-  {
-    V_.P_sfa_.clear();
-    V_.P_stc_.clear();
-
-    for ( size_t i = 0; i < P_.tau_sfa_.size(); i++ )
-    {
-      V_.P_sfa_.push_back( std::exp( -h / P_.tau_sfa_[ i ] ) );
-    }
-    S_.sfa_elems_.resize( P_.tau_sfa_.size(), 0.0 );
+  // initializing adaptation (stc/sfa) variables
+  V_.P_sfa_.clear();
+  V_.P_stc_.clear();
 
-    for ( size_t i = 0; i < P_.tau_stc_.size(); i++ )
-    {
-      V_.P_stc_.push_back( std::exp( -h / P_.tau_stc_[ i ] ) );
-    }
-    S_.stc_elems_.resize( P_.tau_stc_.size(), 0.0 );
+  for ( size_t i = 0; i < P_.tau_sfa_.size(); i++ )
+  {
+    V_.P_sfa_.push_back( std::exp( -h / P_.tau_sfa_[ i ] ) );
+  }
+  S_.sfa_elems_.resize( P_.tau_sfa_.size(), 0.0 );
```

Comment:
I think it is a good idea to let user changes the number of adaptation elements between simulations. As you mentioned, the properties of neuron may change during time. Note that the SFA shape is not exponential and has its own shape. We try to approximate it using exponential elements (mostly two elements). However, it may happen that SFA shape of a neuron changes to a new one, such that it cannot be approximated by two exponential elements anymore and needs more. Therefore, I think it is reasonable that one can change number of SFA elements between simulations.